core/primitives/src/runtime/in_memory_contract.rs

```diff
@@ -0,0 +1,51 @@
+use crate::{hash::CryptoHash, types::AccountId};
+use near_vm_errors::VMError;
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::sync::RwLock;
+
+pub type ContractResult = Result<wasmer_runtime::Module, VMError>;
+pub type InMemoryContractEntry = Option<(CryptoHash, ContractResult)>;
+
+#[derive(Clone)]
+pub struct InMemoryContracts(pub Arc<RwLock<HashMap<AccountId, InMemoryContractEntry>>>);
+
+impl std::fmt::Debug for InMemoryContracts {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "InMemoryContracts")
+    }
+}
+
+impl InMemoryContracts {
+    pub fn new(in_memory_account_ids: &Vec<AccountId>) -> Self {
```

Comment:
```suggestion
    pub fn new(in_memory_account_ids: &[AccountId]) -> Self {
```

`&Vec<T>` is useless type: https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#useless-types. 

`s/&vec![]/&[]/g` after this